src/Terminal/Run.gren

```diff
@@ -0,0 +1,135 @@
+                            , interactive = False
+                            , useColor = False
+                            , stdout = nullStream |> Stream.writable
+                            , stdin = nullStream |> Stream.readable
+                            }
+                            projectOutline
+                    )
+                |> Task.mapError PackageInstallError
+
+        backendCommand outputPath packageResolution =
+            let
+                path =
+                    config.pathToString outputPath
+            in
+            Compiler.Backend.encodeCommand
+                { interactiveSession = False
+                , pathToString = config.pathToString
+                }
+                ( Compiler.Backend.Make
+                    { optimize = False
+                    , sourcemaps = False
+                    , output = Just (Compiler.Backend.Exe path)
+                    , report = Nothing
+                    , projectPath = packageResolution.projectPath
+                    , entryPoints = [ config.moduleName ]
+                    , outline = packageResolution.outline
+                    , rootSources = packageResolution.rootSources
+                    , dependencies = packageResolution.dependencies
+                    }
+                )
+
+
+        compile outputPath encodedCommand_ =
+            Compiler.Backend.run
+                config.cpPermission
+                { useColor = False
+                , compilerPath = config.compilerPath
+                , pathToString = config.pathToString
+                , onInit = 
+                    (\cpOpts ->
+                        config.onBackendInitialized
+                            { backendStreams = cpOpts.streams
+                            , encodedCommand = encodedCommand_
+                            }
+                    )
+                , onComplete = config.onCompiled outputPath
+                }
+    in
+    getOutputPath config.fsPermission
+        |> Task.andThen
+            (\outputPath ->
+                getNullStream
+                    |> Task.andThen installPackages
+                    |> Task.map (backendCommand outputPath)
+                    |> Task.map (compile outputPath)
+            )
+
+
+getOutputPath : FileSystem.Permission -> Task Error Path
+getOutputPath fsPermission =
+    FileSystem.makeTempDirectory fsPermission "gren-run"
+        |> Task.map (Path.append (Path.fromPosixString "app"))
+        |> Task.mapError TempPathError
+
+
+run : RunConfig msg -> Cmd msg
+run config =
+    ChildProcess.spawn config.cpPermission "node" [ config.path ] <|
+        ChildProcess.defaultSpawnOptions config.onInit config.onExit
+
+
+-- not sure if this is the best way to do this?
+getNullStream : Task x (Stream.Transformation Bytes Bytes)
+getNullStream =
+    Stream.customTransformation
+        (\_ _ -> Stream.UpdateState (Bytes.fromString ""))
+        (Bytes.fromString "")
```

Comment:
I think this should work fine. Maybe null streams should be included in `gren-lang/core`? 🤔